### ISSUE TITLE
Specifying a working directory

### ISSUE THREAD

**Issue by wasade** [qiime2/qiime2#221]
It would be nice to be able to specify a temp or working directory within the SDK, and which could be accessed by the subsequent plugin methods (or anything via the SDK). The motivation is a) so that artifacts automatically utilize a different filesystem and b) so that plugins can know what filesystem to use. The specific use case is in HPC environments as they typically have very high performing scratch filesystems, and often times, `/tmp` is limited.

**Comment by jairideout** [comment]
Duplicate of https://github.com/qiime2/qiime2/issues/12. More generally, we plan to implement a configuration system that would include additional settings. This ties into pipeline support so it'll get implemented in parallel when we pick up pipelines again.

**Comment by wasade** [comment]
Okay, thanks. Didn't see the other issue when searching.
